scarb/src/compiler/db.rs

```diff
@@ -184,13 +185,21 @@ fn build_project_config(unit: &CairoCompilationUnit) -> Result<ProjectConfig> {
                 .collect();
 
             // Adds itself to dependencies
-            dependencies.insert(
-                component.package.id.name.to_string(),
-                DependencySettings {
-                    version: (component.package.id.name.to_string() != *CORELIB_CRATE_NAME)
-                        .then_some(component.package.id.version.clone()),
-                },
-            );
+            if component.first_target().kind.is_test() {
```

Comment:
Correct me if I'm wrong, but the title of this PR and previous state of things suggest we only want to filter out integration tests, not all non-tests?

Reply:
Good catch! Submitted a fix :D